### PR TITLE
Make water areas foreground and above linear waterways

### PR DESCRIPTION
**Pull request by pastk**
Sets priorities as following:
- various lines (roads, barriers...) and buildings
- above-water areas (dams, pierses...)
- hatching fills (nature reserves, military zones...)
- area water (lakes, area rivers...)
- linear waterways (rivers, canals...)
- background areas prioritised by bbox area (forests, grass...)

Fixes visual bug with large river area mapped over smaller grass/forest area (introduced by #4581):
forest-over-area-water

Removes river line being displayed over river area (introduced in https://github.com/organicmaps/organicmaps/pull/3803/commits/aa5bccf7648e1914b78ee616bb5df476feb230e5):
waterway-over-water

Roads/paths mapped on a pier / breakwater are displayed now:
roads-over-pierses

### CODE REVIEW

File: map/style_tests/classificator_tests.cpp [hunk 1]
```diff
@@ -235,11 +235,17 @@ UNIT_TEST(Classificator_AreaPriority)
   }, {1, 2, 7, 4}, drule::area);
 
   CheckPriority({
-    // 0
-    {"natural", "water"}, {"landuse", "reservoir"},
-    // 1
+    // 0 - background areas
+    {"landuse", "forest"}, {"landuse", "grass"},
```
Comment (vng) [inline map/style_tests/classificator_tests.cpp:239]: forest and grass are already checked above, no need here

**Review by vng — APPROVED** [review]

**Review by biodranik — APPROVED** [review]
Let's create an issue to show a river/stream direction arrow.

**Comment by pastk** [comment]
Here is a good non-river example, that was fixed (https://www.openstreetmap.org/way/4405171#map=16/52.5064/13.2896):
(the park there has two parts which are smaller than the lake hence the park was rendered on top of the lake)

park-over-lake

**Comment by map-per** [comment]
Tunnels are now coverd by water and not visible anymore

(before (2023.04.02-7-FDroid) and after (2023.05.08-7-Google), https://www.openstreetmap.org/way/33817648)
Screenshot (76)
Screenshot_2023-05-14-09-15-32-153_app organicmaps

**Comment by pastk** [comment]
> Let's create an issue to show a river/stream direction arrow.

https://github.com/organicmaps/organicmaps/issues/5236

**Pull request merged**